parser/gitlabci.go

```diff
@@ -0,0 +1,80 @@
+package parser
+
+import (
+	"fmt"
+
+	"github.com/sethvargo/ratchet/resolver"
+	"gopkg.in/yaml.v3"
+)
+
+type GitLabCI struct{}
+
+// Parse pulls the image references from GitLab CI configuration files.
+// It does not support references with variables.
+
+func (C *GitLabCI) Parse(m *yaml.Node) (*RefsList, error) {
+	var refs RefsList
+	var imageRef *yaml.Node
+
+	// GitLab CI global top level keywords
+	var globalKeywords = map[string]bool{
+		"default":   true,
+		"include":   true,
+		"stages":    true,
+		"variables": true,
+		"workflow":  true,
+	}
```

Comment:
It's more efficient to use a struct instead of a bool, since the empty struct does not allocate

```suggestion
	var globalKeywords = map[string]struct{}{
		"default":   {},
		"include":   {},
		"stages":    {},
		"variables": {},
		"workflow":  {},
	}
```

Reply:
Thanks for comments. I have updated the PR and added example `testdata/gitlabci.yml`.